source/aria.py

```diff
@@ -1,13 +1,60 @@
 # A part of NonVisual Desktop Access (NVDA)
-# Copyright (C) 2009-2019 NV Access Limited, Leonard de Ruijter
+# Copyright (C) 2009-2022 NV Access Limited, Leonard de Ruijter
 # This file is covered by the GNU General Public License.
 # See the file COPYING for more details.
 
-from typing import Dict
+from typing import Dict, Union
 from enum import Enum
 import controlTypes
+from logHandler import log
 
-ariaRolesToNVDARoles: Dict[str, int] = {
+
+def normalizeDetailsRole(detailsRole: Union[str, controlTypes.Role, None]) -> controlTypes.Role:

```

Comment:
I'm not sure this belongs here either.
Also, I'm not sure about the union approach. The code sharing is minimal. The downside is thinking about two very different code paths.

Are there other examples like this?

Reply:
Split for focus and browse mode

Reply:
This function has been removed in favour of handling the 3 cases separately (Firefox focus mode, Chrome Focus mode, Browse mode).